libs/langchain/langchain/chains/combine_documents/map_rerank.py

```diff
@@ -2,17 +2,185 @@
+
+            Returns a dictionary as output that looks like:
+                {
+                    "top_answer": "top-score answer string",
+                    "all_answers": [{"score": ..., "answer": "..."}, {"score": ..., "answer": "..."}, ...]
+                }
+            The scored answers in "all_answers" preserve the order of the input Documents.
+
+
+        Example:
+            .. code-block:: python
+
+                from langchain_community.chat_models import ChatOpenAI
+                from langchain_core.documents import Document
+                from langchain_core.output_parsers import SimpleJsonOutputParser
+                from langchain_core.prompts import ChatPromptTemplate
+                from langchain.chains.combine_documents import create_map_rerank_documents_chain
+
+                llm = ChatOpenAI(model="gpt-3.5-turbo-1106").bind(
+                    response_format={"type": "json_object"}
+                )
+                prompt = ChatPromptTemplate.from_message(
+                    [
+                        ("system", '''Answer the user question using the given context. Score the
+    answer from 1-5 based on how relevant the context is to the question. Return a JSON string with
+    keys 'answer' and 'score.\n\n{context}'''),
+                        ("human", "{question}"),
+                    ]
+                )
+                chain = create_map_rerank_documents_chain(llm, prompt, output_parser=SimpleJsonOutputParser())
+
+                docs = [
+                    Document(page_content="Jesse loves red but not yellow"),
+                    Document(page_content = "Jamal loves green but not as much as he loves orange")
+                ]
+
+                chain.invoke({"context": docs, "question": "Who loves green?"})
+                # -> {
+                #    'all_answers':
+                #       [
+                #           {'answer': 'This context does not provide any information about who loves green.', 'score': 1},
+                #           {'answer': 'Jamal loves green but not as much as he loves orange', 'score': 5}
+                #       ],
+                #   'top_answer': 'Jamal loves green but not as much as he loves orange'
+                # }
+    """  # noqa: E501
+    validate_prompt(prompt, (DOCUMENTS_KEY,))
+    _document_prompt = document_prompt or DEFAULT_DOCUMENT_PROMPT
+    _output_parser = output_parser or (StrOutputParser() | _default_regex)
+
+    format_as_list = partial(
+        format_document_inputs_as_list, document_prompt=_document_prompt
+    )
+
+    answer_chain = prompt | llm | _output_parser
+    answer_chain = answer_chain.with_config(run_name="answer_and_score")
+    map_chain = (format_as_list | answer_chain.map()).with_config(
+        run_name="answer_and_score_all"
+    )
+
+    assign_all_answers: Runnable = RunnableParallel(all_answers=map_chain)
+    assign_all_answers = assign_all_answers.with_config(run_name="assign_all_answers")
```

Comment:
same line as above?

Reply:
done for readability (turns into 4 lines if you do it all in one assignment)